docs/API.md

```diff
@@ -0,0 +1,267 @@
+
+The `App` class is a singleton with static methods.  This is so that it can be used throughout a consuming application merely by importing it:
+
+```
+import { App } from '@edx/frontend-base';
+```
+
+`App` provides a few different methods and properties:
+
+### `App.history`
+
+The application's browser history object.  See the [history documentation](https://github.com/ReactTraining/history/blob/master/docs/GettingStarted.md) for more information.
+
+### `App.authenticatedUser`
+
+A reference to an object containing information about the currently authenticated user.
+
+```
+{
+  userId: <AUTHENTICATED USER's USER ID>,
+  username: <AUTHENTICATED USER's USERNAME>,
+  roles: [<ROLE ONE>, <ROLE TWO>],
+  administrator: <true|false>,
+}
+```
+### `App.decodedAccessToken`
+
+A reference to an object containing the raw, decoded JWT access token.  This is the snake_case source for the data in `App.authenticatedUser`.
+
+### `App.error`
+
+An error object caught by `App.initialize` if an error occurred during initialization.  Useful for error handlers.  Note: This is not populated for errors outside of application initialization.  It's merely a convenience to allow `APP_ERROR` subscribers and `error` overrideHandlers to access the error that put the application in the error phase.
+
+### `App.initialize({ messages, loggingService, overrideHandlers, custom })`
+
+The `App.initialize` method takes an options object with four possible keys:
+
+#### messages
+
+A frontend-i18n-compatible messages object, or an array of such objects.  If an array is provided, duplicate keys are resolved with the last-one-in winning.
+
+#### loggingService
+
+The logging service to be used by @edx/frontend-logging.  It defaults to `NewRelicLoggingService`.
+
+#### overrideHandlers
+
+An optional object of `overrideHandlers` which can be used to replace the default behavior of any part of the startup sequence.  It can also be used to add additional initialization behavior before or after the rest of the sequence.
+
+An example in which we override the authentication handling:
+
+```
+App.initialize({
+  overrideHandlers: {
+    authentication: () => {
+      // As a usage example of overriding one phase of the startup sequence,
+      // providing this function will override the default authentication
+      // initialization.
+    }
+  }
+});
+```
+
+`overrideHandlers` has keys corresponding to the eight lifecycle events.  Including a key will override and replace the corresponding lifecycle handler if provided.  Please see Initialization Lifecycle Phases for more information on the phases responsibilities.
+
+#### custom
+
+The `custom` property can be used to attach custom data to the `App` which will be exposed at `App.custom`.  This data can be used in custom initialization handlers, or elsewhere in the application as necessary.  Note, if you're using this to provide mutable data to the application, _strongly_ consider using React props, context, or Redux instead.
```

Comment:
Can you provide an example of a good use of this?  Or maybe `authenticatedUser` and `error` are good examples of potential uses.  Not sure if it would be better to drop `or elsewhere in the application as necessary`, and just leave this as a useful tool during initialization.